noir-projects/aztec-nr/aztec/src/utils/array/collapse.nr

```diff
@@ -9,12 +9,15 @@ pub fn collapse<T, let N: u32>(input: [Option<T>; N]) -> BoundedVec<T, N>
 where
     T: Eq,
 {
-    // Computing the collpased BoundedVec would result in a very large number of constraints, since we'd need to loop
+    // Computing the collapsed BoundedVec would result in a very large number of constraints, since we'd need to loop
     // over the input array and conditionally write to a dynamic vec index, which is a very unfriendly pattern to the
     // proving backend.
     // Instead, we use an unconstrained function to produce the final collapsed array, along with some hints, and then
     // verify that the input and collapsed arrays are equivalent.
-    let (collapsed, collapsed_to_input_index_mapping) = unsafe { get_collapse_hints(input) };
+    let (collapsed, collapsed_to_input_index_mapping) = unsafe {
+        //@safety The values get verified by the `verify_collapse_hints` function.
+        get_collapse_hints(input)
```

Comment:
The directive being inside the `unsafe` block is super wonky. It also creates a `unsafe { // @safety` mandatory sequence that is just plain bizarre. Discussing this with @TomAFrench to see if we can allow having `@safety` before the block (as is done in Rust) and not have it forced to be inside of it - not a fan of how readability is hindered otherwise.

Reply:
I feel like you are too autistic here.

Reply:
> if you bully Tom enough

bribes work(ed) better.

Reply:
https://github.com/noir-lang/noir/pull/6992

Reply:
Those Argentinians holding together stronk